addons/purchase/models/purchase.py

```diff
@@ -206,7 +213,11 @@ def create(self, vals):
             if 'date_order' in vals:
                 seq_date = fields.Datetime.context_timestamp(self, fields.Datetime.to_datetime(vals['date_order']))
             vals['name'] = self.env['ir.sequence'].with_company(company_id).next_by_code('purchase.order', sequence_date=seq_date) or '/'
-        return super(PurchaseOrder, self).create(vals)
+        vals, partner_vals = self._write_partner_values(vals)
+        res = super(PurchaseOrder, self).create(vals)
+        if partner_vals:
+            res.sudo().write(partner_vals)  # Because the purchase user doesn't have write on `res.partner`
```

Comment:
LGTM. I just curious about why you make this line different from the one in wirte(). Any issue when they are the same?

Reply:
In the write, I can't call myself to avoid recursion error. Then I need to call it in the partner (in the write).
But I should do the same here (call write on the partner_id instead of my self). But it doesn't change anything :/

Reply:
okk. Thanks for the explanation